wsagent/che-core-git-impl-jgit/src/main/java/org/eclipse/che/git/impl/jgit/JGitConnection.java

```diff
@@ -1622,14 +1627,19 @@ protected JSch getJSch(final OpenSshConfig.Host hc, FS fs) throws JSchException
                     String password = remoteUrl.substring(remoteUrl.lastIndexOf(":") + 1, remoteUrl.indexOf("@"));
                     command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(username, password));
                 } else {
+                    String gitUser = request.getAttributes().get("userName");
+                    String gitPassword = request.getAttributes().get("password");
+                    if (gitUser != null && gitPassword != null) {
+                        setCurrentCredentials(gitUser, gitPassword);
+                        requestCredentialsHaveBeenUsed = true;
+                    }
                     credentials = credentialsLoader.getUserCredential(remoteUrl);
                     if (credentials != null) {
-                        command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(credentials.getUserName(),
-                                                                                               credentials.getPassword()));
+                        command.setCredentialsProvider(
```

Comment:
I mean mb it would be better to rewrite it as
```
String gitUser = request.getAttributes().get("userName");
String gitPassword = request.getAttributes().get("password");
if (gitUser != null && gitPassword != null) {
    // set credentials provided in request
    command.setCredentialsProvider(new UsernamePasswordCredentialsProvider(gitUser, gitPassword));
} else {
    // try to find credentials needed by other providers
    credentials = credentialsLoader.getUserCredential(remoteUrl);
    if (credentials != null) {
        command.setCredentialsProvider(
                                new UsernamePasswordCredentialsProvider(credentials.getUserName(), credentials.getPassword()));
    }
}
```
then we'll have next behaviour:
1. If url contains credentials - use it
2. If request contains credentials - use it
3. Try find credentials by credentials loader.
In current state
1. If url contains credentials - use it
2. If request contains set credentials to credentials provider and go to 3
3. Try find credentials by credentials loader. <- It can return our basic credentials or other if any credential provider has credentials for this url. But seems that credentials in request should be setted anyway